### DISCUSSION TITLE
Code snippets as examples for new builders

### DISCUSSION THREAD

**Discussion by JST5000** [XRPLBounties/Proposals#20] [Ideas]
New folks in the community who start building with the XRPL could get a lot of benefit from having simple code snippets which accomplish common goals in a well-written way. 

Most interactions with the ledger tend to follow a similar structure of:
1. Connect to the ledger
2. Create an account/access your account
3. Set up a transaction
4. Sign + send it
5. See what changed (send a request)
6. Close your connection to the ledger

Having more examples of how to do that whole process for various simple use cases of the ledger in a well-documented way would hopefully help new folks ramp up quickly. 

Additionally, the code samples could serve to be open-source references for some of the more nuanced features of the ledger (Ex. [Setting up multisigning](https://xrpl.org/set-up-multi-signing.html#set-up-multi-signing)). 

This is work which would be beneficial to do across all the client libraries ([xrpl.js has some code snippets](https://github.com/XRPLF/xrpl.js/tree/main/packages/xrpl/snippets), but has plenty of gaps)

**Comment by rambodc** [comment]
I agree with you 100%. Most interactions with the ledger follow similar structure. Creating an easy to use web-based visual scripting  editor where users can wire together different features could be very useful. 


![Visual Scripting Editor](https://user-images.githubusercontent.com/43018374/188239762-29d14578-1002-493c-80a4-57cc0b0b9ad2.png)

**Reply by intelliot** [reply]
There is a project working towards this; someone could check in with them to see if they're still making progress. https://xrplgrants.org/grantees?query=Node-RED
